httpserver/lmq_server.cpp

```diff
@@ -128,6 +129,38 @@ void LokimqServer::handle_onion_request(lokimq::Message& message) {
     request_handler_->process_onion_req(std::string(ciphertext), std::string(eph_key), on_response);
 }
 
+bool LokimqServer::check_pn_server_pubkey(const std::string& pk) const {
+    return pk == this->pn_server_key_;
+}
+
+void LokimqServer::handle_notify_add_pubkey(lokimq::Message& message) {
+
+    if (!check_pn_server_pubkey(message.conn.pubkey())) {
+        LOKI_LOG(debug, "Attempt to use notify endpoint by unauthorised pubkey");
+        return;
+    }
+
+    for (const auto &pubkey : message.data) {
+        service_node_->add_notify_pubkey(message.conn, pubkey);
+    }
+
+    lokimq_->send(message.conn, "OK");

```

Comment:
This is a going to end up being a garbage message that the remote lokimq just drops.  If you want to communicate a reply you need to set this up as a request command rather than an ordinary command, and then send a reply back via `message.send_reply(...)`.